tests/linearizability/linearizability_test.go

```diff
@@ -43,77 +45,112 @@ const (
-			failpoint: RandomFailpoint,
+			name: "ClusterOfSize1",
+			failpoint: FailpointConfig{
+				failpoint:           RandomFailpoint,
+				count:               failpointTriggersCount,
+				waitBetweenTriggers: waitBetweenFailpointTriggers,
+			},
 			config: *e2e.NewConfig(
 				e2e.WithClusterSize(1),
 				e2e.WithGoFailEnabled(true),
 				e2e.WithCompactionBatchLimit(100), // required for compactBeforeCommitBatch and compactAfterCommitBatch failpoints
 			),
 		},
 		{
-			name:      "ClusterOfSize3",
-			failpoint: RandomFailpoint,
+			name: "ClusterOfSize3",
+			failpoint: FailpointConfig{
+				failpoint:           RandomFailpoint,
+				count:               failpointTriggersCount,
+				waitBetweenTriggers: waitBetweenFailpointTriggers,
+			},
 			config: *e2e.NewConfig(
 				e2e.WithGoFailEnabled(true),
 				e2e.WithCompactionBatchLimit(100), // required for compactBeforeCommitBatch and compactAfterCommitBatch failpoints
 			),
 		},
 		{
-			name:      "Issue14370",
-			failpoint: RaftBeforeSavePanic,
+			name: "Issue14370",
+			failpoint: FailpointConfig{
+				failpoint:           RaftBeforeSavePanic,
+				count:               failpointTriggersCount,
+				waitBetweenTriggers: waitBetweenFailpointTriggers,
+			},
 			config: *e2e.NewConfig(
 				e2e.WithClusterSize(1),
 				e2e.WithGoFailEnabled(true),
 			),
 		},
+		{
+			name: "Issue14571",
+			failpoint: FailpointConfig{
+				failpoint: MemberAddFailpoint,
+				count:     1,
+			},
+			config: e2e.EtcdProcessClusterConfig{
+				ClusterSize:   1,
+				SnapshotCount: 2,
+			},
+			clientCount: 3,
+			trafficType: AuthTraffic,
+		},
 	}
 	for _, tc := range tcs {
 		t.Run(tc.name, func(t *testing.T) {
-			failpoint := FailpointConfig{
-				failpoint:           tc.failpoint,
-				count:               failpointTriggersCount,
-				waitBetweenTriggers: waitBetweenFailpointTriggers,
+			lg := zaptest.NewLogger(t, zaptest.WrapOptions(zap.AddCaller())).Named(tc.name)
+			if tc.clientCount == 0 {
+				tc.clientCount = 8
+			}
+			if tc.trafficType == nil {
+				tc.trafficType = DefaultTraffic
 			}
 			traffic := trafficConfig{
 				minimalQPS:  minimalQPS,
 				maximalQPS:  maximalQPS,
-				clientCount: 8,
-				traffic:     DefaultTraffic,
+				clientCount: tc.clientCount,
+				traffic:     tc.trafficType,
 			}
-			testLinearizability(context.Background(), t, tc.config, failpoint, traffic)
+			testLinearizability(context.Background(), t, tc.config, tc.failpoint, traffic, lg)
 		})
 	}
 }
 
-func testLinearizability(ctx context.Context, t *testing.T, config e2e.EtcdProcessClusterConfig, failpoint FailpointConfig, traffic trafficConfig) {
+func testLinearizability(ctx context.Context, t *testing.T, config e2e.EtcdProcessClusterConfig, failpoint FailpointConfig, traffic trafficConfig, lg *zap.Logger) {
 	clus, err := e2e.NewEtcdProcessCluster(ctx, t, e2e.WithConfig(&config))
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer clus.Close()
+	if traffic.traffic.AuthEnabled() {
```

Comment:
Setting up authority for traffic should be done by traffic.

Reply:
The initial authority is set up once ideally. 

If moving to `simulateTraffic`, it will be ran by multiple clients.